### ISSUE TITLE
Undefined repository using version 1.58.0 on Bazel 6.3.1

### ISSUE THREAD

**Issue by bendb-instacart** [grpc/grpc-java#10578]
TL;DR: It looks as if there's a missing repository `@com_github_cncf_udpa` in 1.58.0, and on the main branch as well.  Are we making some obvious error integrating grpc-java into our Bazel build?  Would be grateful for advice.

This came up today when bumping our version of grpc-java from 1.56.1 to 1.58.0.

### What version of gRPC-Java are you using?

1.58.0

### What is your environment?

Building on macOS 13.5.2, using JDK 17.

### What did you expect to see?

I expected a successful build.

### What did you see instead?

```
ERROR: /private/var/tmp/_bazel_benbader/a572a5586ef2b44f5b64af9890f651df/external/io_grpc_grpc_java/xds/BUILD.bazel:62:19: no such package '@com_github_cncf_udpa//udpa/type/v1': The repository '@com_github_cncf_udpa' could not be resolved: Repository '@com_github_cncf_udpa' is not defined and referenced by '@io_grpc_grpc_java//xds:xds_protos_java'
ERROR: Analysis of target '//path/to/previously/working:target' failed; build aborted:
```

### Steps to reproduce the bug

Running `bazel build //path/to/previously/building:target` is sufficient for us.

I notice that the only hit the string `udpa` in any of the currently-present Bazel files in this repo is https://github.com/grpc/grpc-java/blob/bc784c0ef9f7727497f8543492979c60d5ce5ef8/xds/BUILD.bazel#L65, which is indeed a reference to the repository that seems to be missing in our build, and in the same target `xds_protos_java` as in the error message.  I couldn't find anywhere in the code here where that repository was instantiated.

We're using the gRPC rules in our workspace thusly:

```
http_archive(
    name = "io_grpc_grpc_java",
    sha256 = "20132d94cd9cc2dbffb1b34d684b00aaa4c0451ecea7f7e8be91eccc9259071f",
    strip_prefix = "grpc-java-1.58.0",
    url = "https://github.com/grpc/grpc-java/archive/v1.58.0.zip",
)

load(
    "@io_grpc_grpc_java//:repositories.bzl",
    "IO_GRPC_GRPC_JAVA_ARTIFACTS",
    "IO_GRPC_GRPC_JAVA_OVERRIDE_TARGETS",
    "grpc_java_repositories",
)

grpc_java_repositories()

# lots of stuff elided

load("@rules_jvm_external//:defs.bzl", "maven_install")

maven_install(
    artifacts = [...] + IO_GRPC_GRPC_JAVA_ARTIFACTS,
    fetch_sources = True,
    generate_compat_repositories = True,
    maven_install_json = "//:maven_install.json",
    override_targets = dict(IO_GRPC_GRPC_JAVA_OVERRIDE_TARGETS.items() +
                            IO_GRPC_GRPC_KOTLIN_OVERRIDE_TARGETS.items()),
    repositories = [
        "https://maven.google.com",
        "https://repo1.maven.org/maven2",
        "https://jitpack.io/",
        "https://packages.confluent.io/maven",
    ],
    strict_visibility = True,
)

load("@maven//:defs.bzl", "pinned_maven_install")

pinned_maven_install()

load("@maven//:compat.bzl", "compat_repositories")

compat_repositories()
```

**Comment by bendb-instacart** [comment]
Hm, can't reproduce this in a toy project.  Googling for some error messages shows some similar-ish problems coming from Gazelle and "naming convention issues".  This is over my head - does Gazelle even come in to play here?  We aren't using it, so far as I can tell.

**Comment by bendb-instacart** [comment]
Well I'm wrong - we _are_ using gazelle, and have some suspiciously-similar repositories therein.  I'm going to close this issue since I'm pretty sure the problem's on our end.
